resolver/resolver_test.go

```diff
@@ -481,3 +297,179 @@ func TestLaunchZK(t *testing.T) {
 		// expected
 	}
 }
+
+func message(asserts ...func(*dns.Msg) []error) func(*dns.Msg) []error {
```

Comment:
`messageAssertions` doesn't give any more information than the context in which this function is used. I had it named `assert` before but fell back to `message` since we're constructing a `message` assertion out of many `assertions` which have type `func(*dns.Msg) []error`